buffer-tests/src/test/java/io/netty/buffer/api/tests/BufferTestSupport.java

```diff
@@ -323,88 +324,91 @@ static void stopExecutor() throws IOException {
     public static void verifyInaccessible(Buffer buf) {
         verifyReadInaccessible(buf);
 
-        verifyWriteInaccessible(buf);
+        verifyWriteInaccessible(buf, BufferClosedException.class, IllegalStateException.class);
 
         try (BufferAllocator allocator = BufferAllocator.heap();
              Buffer target = allocator.allocate(24)) {
-            assertThrows(IllegalStateException.class, () -> buf.copyInto(0, target, 0, 1));
-            assertThrows(IllegalStateException.class, () -> buf.copyInto(0, new byte[1], 0, 1));
-            assertThrows(IllegalStateException.class, () -> buf.copyInto(0, ByteBuffer.allocate(1), 0, 1));
+            assertThrows(UnsupportedOperationException.class, () -> buf.copyInto(0, target, 0, 1));
+            assertThrows(UnsupportedOperationException.class, () -> buf.copyInto(0, new byte[1], 0, 1));
+            assertThrows(UnsupportedOperationException.class, () -> buf.copyInto(0, ByteBuffer.allocate(1), 0, 1));
             if (CompositeBuffer.isComposite(buf)) {
                 assertThrows(IllegalStateException.class, () -> ((CompositeBuffer) buf).extendWith(target.send()));
             }
         }
 
         assertThrows(IllegalStateException.class, () -> buf.split());
         assertThrows(IllegalStateException.class, () -> buf.send());
-        assertThrows(IllegalStateException.class, () -> acquire((ResourceSupport<?, ?>) buf));
-        assertThrows(IllegalStateException.class, () -> buf.copy());
-        assertThrows(IllegalStateException.class, () -> buf.openCursor());
-        assertThrows(IllegalStateException.class, () -> buf.openCursor(0, 0));
-        assertThrows(IllegalStateException.class, () -> buf.openReverseCursor());
-        assertThrows(IllegalStateException.class, () -> buf.openReverseCursor(0, 0));
+        assertThrows(UnsupportedOperationException.class, () -> acquire((ResourceSupport<?, ?>) buf));
+        final RuntimeException c1e = assertThrows(RuntimeException.class, () -> buf.copy());
+        assertTrue(c1e instanceof UnsupportedOperationException || c1e instanceof IllegalStateException);
+        assertThrows(UnsupportedOperationException.class, () -> buf.openCursor());
+        assertThrows(UnsupportedOperationException.class, () -> buf.openCursor(0, 0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.openReverseCursor());
+        assertThrows(UnsupportedOperationException.class, () -> buf.openReverseCursor(0, 0));
```

Comment:
All of these should be `BufferClosedException` as well.

The `split` and `send` methods will need some `isAccessible` checks before their `isOwned` checks.